ksqldb-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -77,22 +77,20 @@ public DataSource getSource(final SourceName sourceName) {
   }
 
   @Override
-  public void putSource(final DataSource dataSource, final boolean allowReplace) {
+  public boolean validateSource(final DataSource dataSource, final boolean allowReplace) {
```

Comment:
We shouldn't be splitting this into two methods, as this makes it much more likely someone will call `putSource` and inadvertently overwrite a source that they shouldn't.

Reply:
That's a good point,
I looked at `DropSourceFactory` , even there 
```
 if (dataSource == null) {
      if (!ifExists) {
        throw new KsqlException("Source " + sourceName.text() + " does not exist.");
      }
    }
```
We throw an Exception which will Fail the command which should not be the outcome right ,

The only place I could find other than `DdlCommandExec` that had capability to pass along a warning was in `KsqlResource.handleKsqlStatements()`
where if an exception is thrown in `MetaStoreImpl` we could possibly catch that and throw a warning like 
```
catch (final KsqlStatementException e) {
      LOG.info("Processed unsuccessfully: " + request + ", reason: " + e.getMessage());
      Errors.badStatement("",e.getSqlStatement(),new KsqlEntityList(
          Arrays.asList(new WarningEntity(e.getSqlStatement(),"Source Exists"))
      ));
```
Even here the first parameter to badStatement() seems it needs to be an empty string , else the msg is printed in the client in `red` . What do you suggest we do

Reply:
> We throw an Exception which will Fail the command which should not be the outcome right ,

That code looks correct to me for dropping a source.  It means if I say `DROP TABLE IF EXISTS FOO` that it won't throw if `FOO` doesn't exist, which is what we want.

As to where to put this new check - you'll need two changes:

1. as above, put a check in `CreateSourceFactory` so that it throws an exception if `IF NOT EXIST` is not present and a source with the same name already exists.  (Following a similar pattern to `DropSourceFactory`).

2. In `DdlCommandExec.executeCreateStream` and `executeCreateTable` you will need to add a check to see if a source with the same name already exists and `isOrReplace()` is false, e.g. 

```java
    @Override
    public DdlCommandResult executeCreateStream(final CreateStreamCommand createStream) {
      final SourceName sourceName = createStream.getSourceName();
      final DataSource existing = metaStore.getSource(sourceName);
      if (existing != null && !createStream.isOrReplace()) {
        return new DdlCommandResult(true, "Stream " + sourceName + " does not exist.");
      }

      final KsqlStream<?> ksqlStream = new KsqlStream<>(
          sql,
          sourceName,
          createStream.getSchema(),
          createStream.getFormats().getOptions(),
          createStream.getTimestampColumn(),
          withQuery,
          getKsqlTopic(createStream)
      );
      
      metaStore.putSource(ksqlStream, createStream.isOrReplace());
      return new DdlCommandResult(true, "Stream created");
    }
```

Reply:
> That code looks correct to me for dropping a source. It means if I say DROP TABLE IF EXISTS FOO that it won't throw if FOO doesn't exist, which is what we want.

@big-andy-coates I always seem to confuse myself with that part :| , so anyway I have made the changes 
i have made changes to `KsqlEngineTest.shouldNotThrowWhenExecutingDuplicateStream()` and `KsqlEngineTest.shouldThrowWhenExecutingDuplicateTable()` , these tests were failing when the query plans for both create source queries had different Kafka topics one had FOO(stream name) and other had BAR, before this PR the test would have failed much before getting to the planning phase, so I have made changes to the queries in the tests , I am unsure if that is alright

Also build in Jenkins has failed I get 404 when I try to open the link , 
When I try `mvn verify` I get 
```
[INFO] [ERROR] Failed to execute goal on project my-first-test-udf: Could not resolve dependencies for project com.acme.ksql.functions:my-first-test-udf:jar:0.1.0-SNAPSHOT: Could not find artifact io.confluent.ksql:ksqldb-udf:jar:6.1.0-SNAPSHOT in confluent (https://packages.confluent.io/maven/) -> [Help 1]
[INFO] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project my-first-test-udf: Could not resolve dependencies for project com.acme.ksql.functions:my-first-test-udf:jar:0.1.0-SNAPSHOT: Could not find artifact io.confluent.ksql:ksqldb-udf:jar:6.1.0-SNAPSHOT in confluent (https://packages.confluent.io/maven/)
```
 and it fails
I am a bit unsure as to how to run all the tests